airbyte-cdk/python/airbyte_cdk/sources/declarative/schema/json_schema.py

```diff
@@ -4,15 +4,22 @@
 
 import json
 import pkgutil
-from dataclasses import InitVar, dataclass
+from dataclasses import InitVar, dataclass, field
 from typing import Any, Mapping, Union
 
+import __main__
 from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
 from airbyte_cdk.sources.declarative.schema.schema_loader import SchemaLoader
 from airbyte_cdk.sources.declarative.types import Config
 from dataclasses_jsonschema import JsonSchemaMixin
 
 
+def _default_file_path() -> str:
+    main_file = __main__.__file__
+    module = main_file.split("/")[-2].replace("-", "_")
+    return f"./{module}/schemas/{{{{options['name']}}}}.json"
```

Comment:
Trying to wrap my head around this one? How come we need the three sets of curly braces around the format str for the name?

Reply:
because it's a f-string (to parameterize it on the module variable)

Reply:
right so the single bracket is on module, and we have another single bracket for `options['name']`, and two `{{` for interpolation, but what is the final `{`?

Reply:
From the docs: https://docs.python.org/3/library/string.html#formatstrings
> If you need to include a brace character in the literal text, it can be escaped by doubling: {{ and }}.

`f"{{hello}}" -> "{hello}"

Here's an example output assuming the module is "source_greenhouse":
`./source_greenhouse/schemas/{{options['name']}}.json`